internal/confirmations/confirmations.go

```diff
@@ -514,7 +514,7 @@ func (bcm *blockConfirmationManager) processBlock(block *BlockInfo) {
 
 			// The block might appear at any point in the confirmation list
 			expectedParentHash := pending.blockHash
-			expectedBlockNumber := pending.blockNumber + 1
+			expectedBlockNumber := pending.blockNumber
```

Comment:
I could do with some context for this one. This change looks wrong on the face of it (and is in code that overall has been used in anger elsewhere), but I might have missed something.

The `+ 1` here is stating the that the first confirmation, will be the block after the thing that's pending.

e.g. if a transaction is in block `1000` then the first confirmation will be block `1001`.

Reply:
Yeah I'm not sure if this is "correct" or not. My observation with mining on demand was that a transaction wouldn't be marked as completed in FireFly until another block was mined. This seemed to be an "off by one" issue, and this did make it work, though like I said I'm not sure it's the "right" solution.